projects/core/src/cart/facade/selective-cart.service.spec.ts

```diff
@@ -0,0 +1,365 @@
+  });
+
+  it('should return loaded false when cart loading', () => {
+    service['cartSelector$'] = of({
+      value: { code: TEST_CART_ID },
+      loading: true,
+      success: false,
+      error: false,
+    });
+
+    let result: boolean;
+    service
+      .getLoaded()
+      .subscribe(value => (result = value))
+      .unsubscribe();
+    expect(result).toEqual(false);
+  });
+
+  it('should not load selective cart for anonymous user', () => {
+    spyOn<any>(service, 'load').and.callThrough();
+    spyOn(multiCartService, 'loadCart').and.stub();
+    spyOn(multiCartService, 'getCartEntity').and.returnValue(
+      of({
+        value: { code: TEST_CART_ID },
+        loading: false,
+        success: false,
+        error: false,
+      })
+    );
+    service['userId'] = OCC_USER_ID_ANONYMOUS;
+    service
+      .getCart()
+      .subscribe()
+      .unsubscribe();
+    expect(service['load']).toHaveBeenCalledTimes(0);
+    expect(multiCartService.loadCart).toHaveBeenCalledTimes(0);
+  });
+
+  it('should return selective cart', () => {
+    spyOn<any>(service, 'load').and.callThrough();
+    spyOn(multiCartService, 'loadCart').and.stub();
+    let result;
+    service
+      .getCart()
+      .subscribe(val => (result = val))
+      .unsubscribe();
+    expect(service['load']).toHaveBeenCalled();
+    expect(result).toEqual({});
+    expect(multiCartService.loadCart).toHaveBeenCalledWith({
+      userId: 'current',
+      cartId: 'selectivecart-test-customer-id',
+    });
+  });
+
+  it('should return cart entries', () => {
+    spyOn(multiCartService, 'getEntries').and.returnValue(of([mockCartEntry]));
+    service
+      .getCart()
+      .subscribe()
+      .unsubscribe();
+    let result;
+    service
+      .getEntries()
+      .subscribe(val => (result = val))
+      .unsubscribe();
+
+    expect(result).toEqual([mockCartEntry]);
+    expect(multiCartService['getEntries']).toHaveBeenCalledWith(
+      'selectivecart-test-customer-id'
+    );
+  });
+
+  it('should load first if cart not loaded before add entry', () => {
+    service['cartSelector$'] = of({
+      loading: false,
+      success: false,
+      error: false,
+    });
+    spyOn(multiCartService, 'addEntry').and.callThrough();
+    spyOn(multiCartService, 'loadCart').and.callThrough();
+    service
+      .getCart()
+      .subscribe()
+      .unsubscribe();
+
+    service.addEntry('productCode', 2);
+    expect(multiCartService['loadCart']).toHaveBeenCalled();
+  });
+  it('should add entry', () => {
+    spyOn(multiCartService, 'addEntry').and.callThrough();
+    service
+      .getCart()
+      .subscribe()
+      .unsubscribe();
+
+    service.addEntry('productCode', 2);
+
+    expect(multiCartService['addEntry']).toHaveBeenCalledWith(
+      OCC_USER_ID_CURRENT,
+      'selectivecart-test-customer-id',
+      'productCode',
+      2
+    );
+  });
+
+  it('should call multiCartService remove entry method with active cart', () => {
```

Comment:
updated.

Reply:
refine testcases for adding 2 entries at the same time

Reply:
We have feature flag(saveForLater) in SaveForLaterModule with feature level 1.5, the recipe is already updated and the sample data PR  https://github.tools.sap/cx-commerce/spartacussampledataaddon/pull/51 is reviewed which will be merged once the code is merged. Since it will be released in 1.5, after it's merged people should consider update the backend to get every functions work .